watertap/examples/flowsheets/electrodialysis/electrodialysis_1stack_conc_recirc.py

```diff
@@ -0,0 +1,495 @@
+    )
+
+    m.fs.pump0 = Pump(property_package=m.fs.properties)
+    m.fs.pump0.del_component("ratioP")
+    m.fs.pump0.del_component("ratioP_calculation")
+    m.fs.pump1 = Pump(property_package=m.fs.properties)
+    m.fs.pump1.del_component("ratioP")
+    m.fs.pump1.del_component("ratioP_calculation")
+
+    # Add electrodialysis (ED) stacks
+    m.fs.EDstack = Electrodialysis1D(
+        property_package=m.fs.properties,
+        operation_mode=ElectricalOperationMode.Constant_Voltage,
+        finite_elements=20,
+        has_pressure_change=True,
+        has_nonohmic_potential_membrane=True,
+        has_Nernst_diffusion_layer=True,
+        limiting_current_density_method=LimitingCurrentDensityMethod.Theoretical,
+        pressure_drop_method=PressureDropMethod.Darcy_Weisbach,
+        hydraulic_diameter_method=HydraulicDiameterMethod.spacer_specific_area_known,
+        friction_factor_method=FrictionFactorMethod.Gurreri,
+    )
+    m.fs.sepa1 = Separator(
+        property_package=m.fs.properties,
+        outlet_list=["to_disp", "to_conc_in1"],
+    )
+    m.fs.prod = Product(property_package=m.fs.properties)
+    m.fs.disp = Product(property_package=m.fs.properties)
+
+    # Touching needed variables for initialization and displaying results
+    m.fs.feed.properties[0].conc_mass_phase_comp[...]
+    m.fs.sepa0.to_dil_in_state[0].conc_mass_phase_comp[...]
+    m.fs.sepa0.to_conc_in0_state[0].conc_mass_phase_comp[...]
+    m.fs.sepa1.to_disp_state[0].conc_mass_phase_comp[...]
+    m.fs.sepa1.to_conc_in1_state[0].conc_mass_phase_comp[...]
+    m.fs.mix0.from_feed_state[0].conc_mass_phase_comp[...]
+    m.fs.mix0.from_conc_out_state[0].conc_mass_phase_comp[...]
+    m.fs.pump0.control_volume.properties_in[0].conc_mass_phase_comp[...]
+    m.fs.pump0.control_volume.properties_in[0].conc_mass_phase_comp[...]
+    m.fs.pump1.control_volume.properties_in[0].conc_mass_phase_comp[...]
+    m.fs.pump1.control_volume.properties_in[0].conc_mass_phase_comp[...]
+    m.fs.prod.properties[0].conc_mass_phase_comp[...]
+    m.fs.disp.properties[0].conc_mass_phase_comp[...]
+
+    m.fs.feed.properties[0].flow_vol_phase[...]
+    m.fs.sepa0.to_dil_in_state[0].flow_vol_phase[...]
+    m.fs.sepa0.to_conc_in0_state[0].flow_vol_phase[...]
+    m.fs.sepa1.to_disp_state[0].flow_vol_phase[...]
+    m.fs.sepa1.to_conc_in1_state[0].flow_vol_phase[...]
+    m.fs.mix0.from_feed_state[0].flow_vol_phase[...]
+    m.fs.mix0.from_conc_out_state[0].flow_vol_phase[...]
+    m.fs.pump0.control_volume.properties_in[0].flow_vol_phase[...]
+    m.fs.pump0.control_volume.properties_in[0].flow_vol_phase[...]
+    m.fs.pump1.control_volume.properties_in[0].flow_vol_phase[...]
+    m.fs.pump1.control_volume.properties_in[0].flow_vol_phase[...]
+    m.fs.prod.properties[0].flow_vol_phase[...]
+    m.fs.disp.properties[0].flow_vol_phase[...]
+    m.fs.EDstack.diluate.properties[...].flow_vol_phase[...]
+    m.fs.EDstack.concentrate.properties[...].flow_vol_phase[...]
```

Comment:
I saw something similar in #875 .. why are we doing this exactly? The lazy property creation should do this for you, and calculating additional properties not strictly needed to determine a feasible/optimal solution can cause additional numerical difficulties for Ipopt (or any solver).

Reply:
> I saw something similar in #875 .. why are we doing this exactly? The lazy property creation should do this for you, and calculating additional properties not strictly needed to determine a feasible/optimal solution can cause additional numerical difficulties for Ipopt (or any solver).

Hi Ben, could you explain what is "lazy property creation"?  So these are all properties that are later needed and touching them here seem to provide  some pre-calculation to set some better initial values.  I don't think they cause difficulties here but probably removing them will also be fine.

Reply:
Basically they're only created after they're accessed the first time. If there is some benefit to doing this here as opposed to later on, then maybe there's something that needs to be done when we create the properties initially.

I'm guessing you do this because the initial values for the properties are better aligned. I wonder if the property constructor methods should calculate the value based on the existing variable values, e.g., call `calculate_variable_from_constraint` to get an initial value for the property.

@andrewlee94 I'm sure there's probably a good reason this isn't the standard thing to do. Thoughts?

Reply:
Upon discussion with @TimBartholomew , I removed unnecessary touching of variables and kept minimum needed for reporting outcome. I recommend accepting this change for now, while we can still linger on Ben's comment above and make subsequent changes if needed.